### ISSUE TITLE
[DISCUSSION] Add a GitHub Action "Variable Threshold Pylint" to the Repo.

### ISSUE THREAD

**Issue by vybhav72954** [HarshCasper/Rotten-Scripts#495]
# Description

While going through an [Issue](https://github.com/HarshCasper/Rotten-Scripts/pull/477) I ran into a comment by @HarshCasper about `PyLint` scores. I think we can introduce a custom PyLint action in the Repo.

Although there are similar actions in the marketplace. I believe as this is a first-timer friendly repository. We can't really expect everyone, for example, to write `docstring` for `__doc__` attribute in python. But having a lower score check will also ensure that styling and basic guidelines of `pep-8` are met.

This would be beneficial for new programmers and the reviewers as well. 
***Who doesn't like a well-formatted code?***

As this is a rather big change, I suggest it is discussed before anything. 

## Language

- [ ] Python

## Type of issue

- [ ] Feature

## Checklist:

- **For Feature**

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| GitHub PyLint Action | To avoid spammy PR and maintain the aesthetics of the repo. | feature | @vybhav72954  |

**Is your feature request related to a problem? Please describe.**: PyLint scores provide basic clarity about how well the code looks, Implementing a GitHub action that will ensure that spammy PR or badly written code doesn't need a review in the first place, will save the time of moderators and also promote healthy coding practices.

**Describe the solution you'd like**: A PyLint action with a modifiable check so it is suitable for first-timers but at the same time can Fail spammy PRs.

**Comment by vybhav72954** [comment]
@HarshCasper @srimani-programmer @seema1711 I need some help with Action.

I have made the `.yml` file. and a `pylint.py` file.

Am not sure how to make sure that my workflow is triggered only for the new commits. Can you guide me? I can send a request to you guys from my private repo too.

**Comment by vybhav72954** [comment]
[logs_88.zip](https://github.com/HarshCasper/Rotten-Scripts/files/5451677/logs_88.zip)

@HarshCasper These are log files generated from the Action. I will also upload a few screenshots soon. Details can be found in the logs.

**Comment by vybhav72954** [comment]
@HarshCasper I have finalized the Action from my side.

Few things I just wanted to confirm/clear before starting a PR:

- Right now the Function is very strict as I was unsure how to make it less strict, I have set the threshold at a score of 1, the Action won't fail for badly structured code, but will give them suggestions, in case the code is really bad (something below 1), the action would fail.

- I have found a way around PyLint using which I can make code a bit more lenient, (It is basically like ignoring the possible errors). This method is although not recommendable, but can be adopted considering we can't expect everyone to come up with flawless code and nitpicking can be avoided.

- Secondly, Code if fails for a particular code in the PR, won't run for the next code, I tried really hard to put this in a loop and `continue` even if there is a failure, YAML doesn't provide this feature.

- I will ask about this in Git Comunity and Stack Overflow, but I still believe that the action is ready for a Pilot run.

- I am quite sure that there is no such action available right now, (the article on `medium` is not flexible at all), so it would take a few pilot tuns on the Repo to be completely optimized, (hope you understand). 

I have made sure that the code is error-free, and I have explained the functionality above, in case you want an addition.
